src/components/VerificationCode/VerificationCode.js

```diff
@@ -0,0 +1,292 @@
+      // If that is the case, we focus the first input
+      // We use a timeout to take the execution of this action out of the event loop so that the browser doesn't focus the body
+      setTimeout(() => {
+        this.digitInputNodes[0].focus()
+      }, 0)
+    } else if (!target.classList.contains('DigitInput')) {
+      // If the click happened anywhere inside but not an input, the browser tries to focus the body (or a parent), we bring the focus
+      // back to the cached activeElement using a timeout as above
+      setTimeout(() => {
+        activeElement.focus()
+      }, 0)
+    }
+  }
+
+  handleInputKeyUp = (index, e) => {
+    const { key } = e
+
+    if (key !== 'Meta') {
+      const { value } = e.target
+      const { numberOfChars, onChange } = this.props
+      const digitMask = this.digitMaskNodes[index]
+
+      /* istanbul ignore else */
+      if (key === 'Backspace') {
+        let selectionText = getCleanSelectedText()
+
+        if (selectionText.length > 1) {
+          clearAll(this.digitInputNodes, this.digitMaskNodes)
+          showInputDigits(this.digitInputNodes, this.digitMaskNodes)
+          this.digitInputNodes[0].focus()
+          onChange('')
+        } else if (value === '' && !digitMask.innerText) {
+          // Tested  ¯\_(ツ)_/¯
+          /* istanbul ignore next */
+          const prevIndex = index === 0 ? 0 : index - 1
+          const previousDigit = this.digitInputNodes[prevIndex]
+
+          digitMask.innerText = value
+          previousDigit && previousDigit.select()
+          onChange(getCurrentCodeValue(this.digitInputNodes))
+        } else {
+          digitMask.innerText = value
+          onChange(getCurrentCodeValue(this.digitInputNodes))
+        }
+      } else if (key === 'ArrowLeft') {
+        const prevIndex = index === 0 ? 0 : index - 1
+        const previousDigit = this.digitInputNodes[prevIndex]
+
+        previousDigit && previousDigit.select()
+      } else if (key === 'ArrowRight') {
+        const nextIndex =
+          index === numberOfChars - 1 ? numberOfChars : index + 1
+        const nextDigit = this.digitInputNodes[nextIndex]
+
+        nextDigit && nextDigit.select()
+      } else if (key.length === 1) {
+        // all chars have a length of 1, special keys have more
+        const nextIndex =
+          index === numberOfChars - 1 ? numberOfChars : index + 1
+        const nextDigit = this.digitInputNodes[nextIndex]
+
+        digitMask.innerText = value
+        nextDigit && nextDigit.select()
+
+        onChange(getCurrentCodeValue(this.digitInputNodes))
+      }
+    }
+  }
+
+  handleInputClick = e => {
+    showInputDigits(this.digitInputNodes, this.digitMaskNodes)
+    e.target.select()
+  }
+
+  render() {
+    const { numberOfChars, isValid } = this.props
+
+    return (
+      <VerificationCodeFieldUI
+        className={this.getClassName()}
+        innerRef={this.setVerificationCodeFieldNode}
```

Comment:
will need to be rename to `ref` once it's merged in v3. 

innerRef will be use for our component (otherwise it creates some conflicts with react 🤷‍♂), and ref is the keyword needed when set on html node

Also we could declare the ref here with React.createRef(). Then you would not need the `setVerificationCodeFieldNode` method.

Reply:
Yes, I started using `ref` because I remembered, but then the console reminded _me_ that fancy does not support them so I switched them